### PR TITLE
Add minor version release template

### PR DESCRIPTION
**Pull request by leahwicz**
Adding a new issue template for releasing new minor versions so we don't miss out on any steps

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/minor-version-release.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+labels: ''
+assignees: ''
+
+---
+
+### Engineering TODOs:
+- [ ] dbt-release workflow 
+- [ ] Create new protected `x.latest` branch 
+- [ ] Create a platform issue to update dbt Cloud
+- [ ] Generate schema updates
```
Comment (jtcohen6) [inline .github/ISSUE_TEMPLATE/minor-version-release.md:14]: Let's move this to be first, ahead of `dbt-release` workflow
```suggestion
- [ ] Determine if schemas have changed. If so, generate new schemas and push to schemas.getdbt.com
```

File: .github/ISSUE_TEMPLATE/minor-version-release.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+- [ ] Bump plugin versions (dbt-spark + dbt-presto), add compatibility as needed
+   - [ ]  Spark 
+   - [ ]  Presto
+- [ ] Create a platform issue to update dbt-spark versions to dbt Cloud 
+
+### Product TODOs:
+- [ ] Merge PRs into next.docs.getdbt.com
```
Comment (jtcohen6) [inline .github/ISSUE_TEMPLATE/minor-version-release.md:21]:
```suggestion
- [ ] Finalize migration guide (next.docs.getdbt.com)
```

File: .github/ISSUE_TEMPLATE/minor-version-release.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+- [ ] Release new minor version of dbt-utils with new version compatibility
```
Comment (jtcohen6) [inline .github/ISSUE_TEMPLATE/minor-version-release.md:22]:
```suggestion
- [ ] Release new version of dbt-utils with new dbt version compatibility. If there are breaking changes requiring a minor version, plan upgrades of other packages that depend on dbt-utils.
```

File: .github/ISSUE_TEMPLATE/minor-version-release.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+- [ ] Publish discourse (draft here)
```
Comment (jtcohen6) [inline .github/ISSUE_TEMPLATE/minor-version-release.md:23]:
```suggestion
- [ ] Publish discourse
```

**Review by jtcohen6 — COMMENTED** [review]
Some of these TODOs are intersecting. For instance, I wait to announce in Discourse/Slack until the new versions are available in dbt Cloud. I don't know if the separation between Engineering + Product makes sense, or if we should break it up into sections like:
- Release Core
- Release Cloud
- Announce
- Post-release (which should also include the creation of a tracking issue like #3441)

**Comment by leahwicz** [comment]
> Some of these TODOs are intersecting...

I wouldn't mind putting it in sections like that and then it will be in logical order. I can still label the items as engineering/product so we know who is going to do what

**Comment by leahwicz** [comment]
@jtcohen6 let me know if I divided those sections up correctly

**Review by jtcohen6 — APPROVED** [review]
I really like it!

A subset of these things (about half) are relevant for patch releases as well. I think we could start with this checklist for those as well, and trim it down accordingly.

**Pull request merged**